src/overlays/actors/ovl_En_Rd/z_en_rd.h

```diff
@@ -8,29 +8,75 @@ struct EnRd;
 typedef void (*EnRdActionFunc)(struct EnRd*, GlobalContext*);
 
+#define EN_RD_GET_FLAGS(thisx) (((thisx)->params & 0xFF00) >> 8)
+
+typedef enum GibdoLimbs {
+    /*  0 */ REDEAD_GIBDO_LIMB_NONE,
+    /*  1 */ REDEAD_GIBDO_LIMB_ROOT,
+    /*  2 */ REDEAD_GIBDO_LIMB_LEFT_LEG_ROOT,
+    /*  3 */ REDEAD_GIBDO_LIMB_LEFT_THIGH,
+    /*  4 */ REDEAD_GIBDO_LIMB_LEFT_SHIN,
+    /*  5 */ REDEAD_GIBDO_LIMB_LEFT_ANKLE,
+    /*  6 */ REDEAD_GIBDO_LIMB_LEFT_FOOT,
+    /*  7 */ REDEAD_GIBDO_LIMB_RIGHT_LEG_ROOT,
+    /*  8 */ REDEAD_GIBDO_LIMB_RIGHT_THIGH,
+    /*  9 */ REDEAD_GIBDO_LIMB_RIGHT_SHIN,
+    /* 10 */ REDEAD_GIBDO_LIMB_RIGHT_ANKLE,
+    /* 11 */ REDEAD_GIBDO_LIMB_RIGHT_FOOT,
+    /* 12 */ REDEAD_GIBDO_LIMB_UPPER_BODY_ROOT,
+    /* 13 */ REDEAD_GIBDO_LIMB_UPPER_BODY,
+    /* 14 */ REDEAD_GIBDO_LIMB_TORSO,
+    /* 15 */ REDEAD_GIBDO_LIMB_LEFT_ARM_ROOT,
+    /* 16 */ REDEAD_GIBDO_LIMB_LEFT_SHOULDER_AND_UPPER_ARM,
+    /* 17 */ REDEAD_GIBDO_LIMB_LEFT_FOREARM,
+    /* 18 */ REDEAD_GIBDO_LIMB_LEFT_HAND,
+    /* 19 */ REDEAD_GIBDO_LIMB_RIGHT_ARM_ROOT,
+    /* 20 */ REDEAD_GIBDO_LIMB_RIGHT_SHOULDER_AND_UPPER_ARM,
+    /* 21 */ REDEAD_GIBDO_LIMB_RIGHT_FOREARM,
+    /* 22 */ REDEAD_GIBDO_LIMB_RIGHT_HAND,
+    /* 23 */ REDEAD_GIBDO_LIMB_HEAD_ROOT,
+    /* 24 */ REDEAD_GIBDO_LIMB_HEAD,
+    /* 25 */ REDEAD_GIBDO_LIMB_PELVIS,
+    /* 26 */ REDEAD_GIBDO_LIMB_MAX
+} RedeadGibdoLimbs;
+
+typedef enum {
+    /* -3 */ EN_RD_TYPE_GIBDO_RISING_OUT_OF_COFFIN = -3,
+    /* -2 */ EN_RD_TYPE_GIBDO = -2,
+    /* -1 */ EN_RD_TYPE_DOES_NOT_MOURN = -1,
+    /*  0 */ EN_RD_TYPE_DOES_NOT_MOURN_IF_WALKING = 0,
+    /*  1 */ EN_RD_TYPE_REGULAR,
+    /*  2 */ EN_RD_TYPE_CRYING,
+    /*  3 */ EN_RD_TYPE_INVISIBLE
+} EnRdType;
+
 typedef struct EnRd {
     /* 0x0000 */ Actor actor;
     /* 0x014C */ Vec3s firePos[10];
     /* 0x0188 */ SkelAnime skelAnime;
-    /* 0x01CC */ Vec3s jointTable[26];
-    /* 0x0268 */ Vec3s morphTable[26];
-    /* 0x0304 */ u8 unk_304;
-    /* 0x0305 */ u8 unk_305;
-    /* 0x0306 */ u8 unk_306;
-    /* 0x0307 */ u8 unk_307;
+    /* 0x01CC */ Vec3s jointTable[REDEAD_GIBDO_LIMB_MAX];
+    /* 0x0268 */ Vec3s morphTable[REDEAD_GIBDO_LIMB_MAX];
+    /* 0x0304 */ u8 grabState;
+    /* 0x0305 */ u8 isMourning;
+    /* 0x0306 */ u8 playerStunWaitTimer; // Cannot stun the player if this is non-zero
+    /* 0x0307 */ u8 grabWaitTimer; // Cannot grab the player if this is non-zero
     /* 0x0308 */ EnRdActionFunc actionFunc;
-    /* 0x030C */ s16 unk_30C;
-    /* 0x030E */ s16 unk_30E;
-    /* 0x0310 */ s16 unk_310;
-    /* 0x0312 */ s16 unk_312;
-    /* 0x0314 */ s16 unk_314;
-    /* 0x0316 */ s16 unk_316;
-    /* 0x0318 */ u8 unk_318;
-    /* 0x0319 */ u8 unk_319;
-    /* 0x031A */ u8 unk_31A;
-    /* 0x031B */ u8 unk_31B;
-    /* 0x031C */ u8 unk_31C;
-    /* 0x031D */ u8 unk_31D;
+    /* 0x030C */ union {
+                    s16 animationJudderTimer;
+                    s16 deathTimer;
+                    s16 coffinRiseForwardAccelTimer;
+                };
```

Comment:
I think this should just be a generic `s16 timer`, as sad as it is
`union` doesn't seem to get much use in oot code, and also I think it's just confusing to use (accidentally use the same memory for two purposes)

Reply:
In this specific case, I'd be relatively okay with just doing a single `s16 timer`. But I want to gently challenge what you said about `union`, because the devs *did* actually accidentally or purposefully use the same memory for multiple different purposes. Looking at `Obj_Tokeidai` in MM for example, there's a single struct variable that's used as a timer, as a translation amount, as acceleration, and as speed, all depending on the context. You can see another example in OoT with En_Tite, where actionVar1 is used as two different timers and as a state variable (the "actionVar + defines" thing that actor does is just a different way of implementing a union). Additionally, there are still lots of undocumented or semi-documented code in both codebases, which can make it hard to see *when* they're using the same memory for different purposes; it all looks the same when it's unk_XXX.

I agree that it can sort of be confusing to have multiple different variables that end up being backed by the same bit of memory, but there are clear cases where it's obvious that's what they did. This specific case *isn't* one of those clear cases, since yeah, they're all just timers. But IMO, the benefits of labeling the timers individually outweigh the drawbacks. Feel free to disagree, though.

Reply:
I would be way less on the fence with this if there weren't other timers outside the union like `sunsSongStunTimer`, which make it more of a pain to remember which timers are in the union

Even in the spots where a union makes sense because a timer is reused as a non-timer-like value, I'm still doubting a union was used originally, because it really saves only bytes of memory (nearly nothing), and if they had been worried about memory usage that much there was a lot of more important things to change that would matter way more

That said I do agree it's nicer for reading to know what timer is being used. So if others are ok with it I'd rather see something like:

```suggestion
    /* 0x030C */ union {
                    s16 animationJudder;
                    s16 death;
                    s16 coffinRiseForwardAccel;
                } timer;
```

which rules out confusion about which timers are in the union imo

Reply:
> I'm still doubting a union was used originally, because it really saves only bytes of memory (nearly nothing), and if they had been worried about memory usage that much there was a lot of more important things to change that would matter way more

We can't really say for certain what they were thinking. Maybe one dev really cared about the memory usage of their actor. Even if the engine as a whole is inefficient with memory, that doesn't necessarily rule out that an individual piece of code might not be.

Yeah, I see what you mean about it being confusing to remember what timers are in the union. MM's version of Rd adds even *more* timers, including one that shares the same memory as the rotational velocity for one of the Redead's dances (this is what prompted me to start using `union` in the first place). I am totally fine with what you proposed here, so long as others are okay with it too.